misc/testerina/modules/testerina-runtime/src/main/java/org/ballerinalang/test/runtime/util/TesterinaUtils.java

```diff
@@ -198,12 +198,38 @@ private static Test getTest(String testName, List<Test> tests) {
     /**
      * Provides the updated test functions for single Execution after adding the dependent tests.
      *
-     * @param currentTests the tests available in current test suite
-     * @param functions    the list of test functions provided by user
+     * @param suite         current test suite
+     * @param functions     the list of test functions provided by user
      * @return updated list of test functions
      */
-    public static List<Test> getSingleExecutionTests(List<Test> currentTests, List<String> functions) {
-        List<String> updatedFunctionList = getUpdatedFunctionList(currentTests, functions);
+    public static List<Test> getSingleExecutionTests(TestSuite suite, List<String> functions) {
+        List<String> filteredList = new ArrayList<>();
+
+        // Go through each function in the functionsList
+        for (String function : functions) {
+            if (function.contains(":")) {
+                String[] functionDetail = function.split(":");
+
+                if (functionDetail[0].equals(suite.getPackageID())) {
+                    if (functionDetail[1].equals(TesterinaConstants.WILDCARD)) {
```

Comment:
Don't we need to handle the potential `IndexOutOfBoundException`(s) in here?

Reply:
Yes. I had missed this. I have handled the exception and now it throws an error.

Reply:
Great!